### PR TITLE
Bump Phpstan to level 4

### PR DESCRIPTION
**Pull request by VincentLanglet**

### CODE REVIEW

File: src/Error/Error.php [hunk 1]
```diff
@@ -142,11 +142,7 @@ private function updateRepr(): void
         }
 
         if ($this->name) {
-            if (\is_string($this->name) || $this->name instanceof \Stringable) {
-                $name = \sprintf('"%s"', $this->name);
-            } else {
-                $name = json_encode($this->name);
-            }
```
Comment (VincentLanglet) [inline src/Error/Error.php:149]: This is weird.

The property is typed as `private ?string $name;` so it always a string in this `if`.

File: src/Extension/CoreExtension.php [hunk 2]
```diff
@@ -1199,7 +1199,7 @@ public static function length(string $charset, $thing): int
             return mb_strlen($thing, $charset);
         }
 
-        if (is_countable($thing) || $thing instanceof \SimpleXMLElement) {
```
Comment (VincentLanglet) [inline src/Extension/CoreExtension.php:1202]: SimpleXMLElement is countable.

File: src/Parser.php [hunk 3]
```diff
@@ -362,7 +365,7 @@ private function filterBodyNodes(Node $node, bool $nested = false): ?Node
         // we need to discard the wrapping "Node" for the "body" node
         $nested = $nested || !$node instanceof Nodes;
         foreach ($node as $k => $n) {
-            if (null !== $n && null === $this->filterBodyNodes($n, $nested)) {
```
Comment (VincentLanglet) [inline src/Parser.php:365]: node is `@implements \IteratorAggregate<int|string, Node>` so `$n` is never null.

File: src/Extension/CoreExtension.php (hunk 2 above)
Comment (fabpot) [reply src/Extension/CoreExtension.php:1202]: Since 8.0. As we bumped the min version, this change is valid.

File: src/Error/Error.php [hunk 4]
```diff
@@ -142,11 +142,7 @@ private function updateRepr(): void
...
+            $name = \sprintf('"%s"', $this->name);
```
Comment (fabpot) [inline src/Error/Error.php]: As $name is not used anywhere else, maybe we can merge this line with the next?

File: src/Error/Error.php (hunk 1 above)
Comment (fabpot) [reply src/Error/Error.php:149]: I've just had a quick look to see why this is done this way and this is indeed not valid anymore.

File: src/Error/Error.php (hunk 4 above)
Comment (VincentLanglet) [reply src/Error/Error.php]: Yes, Done in https://github.com/twigphp/Twig/pull/4494/commits/8a2f2c0c1128831dd50a12b4d926f1f34ee8e693

**Comment by fabpot** [comment]
Thank you @VincentLanglet.

**Pull request merged**